src/components/PeoplePage.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { useEffect, useState } from 'react';
+import { Loader } from './Loader';
+import { getPeople } from '../api';
+import { Person } from '../types';
+import { PersonLink } from './PersonLink';
+import { NotFoundPage } from './NotFoundPage';
+
+export const PeoplePage: React.FC = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState(false);
+
+  useEffect(() => {
+    setLoading(true);
+
+    getPeople()
+      .then(response => setPeople(response))
+      .catch(() => setError(true))
+      .finally(() => setLoading(false));
+  }, []);
+
+  const findParent = (parentName: string) => {
+    return people.find(person => person.name === parentName);
+  };
+
+  const namesFromServer = people.map(person => person.name);
+
+  return (
+    <>
+      <h1 className="title">People Page</h1>
+      {error && <NotFoundPage />}
```

Comment:
Displaying the `NotFoundPage` component when there is an error might not be appropriate if the error is related to data fetching. Consider using a more specific error message or component to indicate a data fetching error, rather than a 'Page not found' message.